### PR TITLE
MINOR: Add test for describe topic with ID

### PR DESCRIPTION
**Pull request by ex172000**
Increase the coverage for describeTopics method in the AdminClient

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ex172000** [comment]
The test failed, but not related. All failed tests are integration test and this is a unit test only change.

**Comment by ex172000** [comment]
Hi @divijvaidya, any chance you can check this commit? Thank you in advance!

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java [hunk 1]
```diff
@@ -2354,6 +2354,55 @@ public void testDeleteRecords() throws Exception {
         }
     }
 
+    @Test
+    public void testDescribeTopicsByIds() {
+        try (AdminClientUnitTestEnv env = new AdminClientUnitTestEnv(mockCluster(4, 0),
+                AdminClientConfig.RETRIES_CONFIG, "2")) {
```
Comment (divijvaidya) [inline clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java]: Just use `try (AdminClientUnitTestEnv env = mockClientEnv()) {` maybe? Setting cluster to 4 nodes and 2 retries does't sound relevant to this test

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java [hunk 2]
```diff
@@ -2354,6 +2354,55 @@ public void testDeleteRecords() throws Exception {
...
+                            env.cluster().nodes(),
+                            env.cluster().clusterResource().clusterId(),
+                            env.cluster().controller().id(),
+                            singletonList(new MetadataResponse.TopicMetadata(Errors.NONE, topicName, topicId, false,
+                                    singletonList(partitionMetadata), MetadataResponse.AUTHORIZED_OPERATIONS_OMITTED))));
+            TopicCollection.TopicIdCollection topicIds = TopicCollection.ofTopicIds(
+                    singletonList(topicId));
+            try {
+                DescribeTopicsResult result = env.adminClient().describeTopics(
+                        TopicCollection.ofTopicIds(singletonList(topicId)));
```
Comment (divijvaidya) [inline clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java]: could we use `topicIds` local variable defined above at line 2382?

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java [hunk 3]
```diff
@@ -2354,6 +2354,55 @@ public void testDeleteRecords() throws Exception {
...
+            }
+
+            // Invalid ID
+            try {
+                DescribeTopicsResult result = env.adminClient().describeTopics(
+                        TopicCollection.ofTopicIds(singletonList(Uuid.ZERO_UUID)));
+                result.allTopicIds().get();
+                fail("describe with Uuid.ZERO_UUID should throw exception");
+            } catch (Exception e) {
+                assertEquals("The given topic id 'AAAAAAAAAAAAAAAAAAAAAA' cannot be represented in a request.",
```
Comment (divijvaidya) [inline clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:2419]: please also assert exception type to be `InvalidTopicException`

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java [hunk 4]
```diff
@@ -2354,6 +2354,55 @@ public void testDeleteRecords() throws Exception {
         }
     }
 
+    @Test
+    public void testDescribeTopicsByIds() {
+        try (AdminClientUnitTestEnv env = new AdminClientUnitTestEnv(mockCluster(4, 0),
+                AdminClientConfig.RETRIES_CONFIG, "2")) {
+            env.kafkaClient().setNodeApiVersions(NodeApiVersions.create());
+
+            // Valid ID
```
Comment (divijvaidya) [inline clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:2362]: we have added a case for valid Id, another for unrepresentable id and perhaps we have have one more for id which does not exist in the broker? Can you please add the third case?

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java [hunk 5]
```diff
@@ -2354,6 +2354,55 @@ public void testDeleteRecords() throws Exception {
...
+                assertEquals(topicName, allTopicIds.get(topicId).name());
+            } catch (Exception e) {
+                fail("describe with valid topicId should not fail", e);
+            }
+
+            // Invalid ID
+            try {
+                DescribeTopicsResult result = env.adminClient().describeTopics(
+                        TopicCollection.ofTopicIds(singletonList(Uuid.ZERO_UUID)));
+                result.allTopicIds().get();
```
Comment (divijvaidya) [inline clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:2416]: you can also choose to use `TestUtils.assertFutureError`

**Comment by ex172000** [comment]
Thank you @divijvaidya for the timely review, updated!

**Comment by ex172000** [comment]
Hi @divijvaidya can you take another look?

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java [hunk 6]
```diff
@@ -2354,6 +2354,75 @@ public void testDeleteRecords() throws Exception {
...
+            }
+
+            // ID not exist in brokers
+            Uuid nonExistID = Uuid.randomUuid();
+            env.kafkaClient().prepareResponse(RequestTestUtils
+                    .metadataResponse(
+                            env.cluster().nodes(),
+                            env.cluster().clusterResource().clusterId(),
+                            env.cluster().controller().id(),
+                            asList()));
```
Comment (dengziming) [inline clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java]: It's better to use emptyList()

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java [hunk 7]
```diff
@@ -2354,6 +2354,75 @@ public void testDeleteRecords() throws Exception {
...
+            try {
+                DescribeTopicsResult result = env.adminClient().describeTopics(
+                        TopicCollection.ofTopicIds(singletonList(nonExistID)));
+                TestUtils.assertFutureError(result.allTopicIds(), InvalidTopicException.class);
+                result.allTopicIds().get();
+                fail("describe with non-exist topic ID should throw exception");
+            } catch (Exception e) {
+                assertEquals(
+                        String.format("org.apache.kafka.common.errors.InvalidTopicException: TopicId %s not found.", nonExistID),
```
Comment (dengziming) [inline clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:2407]: I think we are supposed to get a UnknownTopicIdException in this case, but it's not related to this PR.

**Review by dengziming — APPROVED** [review]
LGTM, just a minor comment.

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java (hunk 7 above)
Comment (ex172000) [reply clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:2407]: Thanks for pointing out, I can make a follow up change looking into this.

**Review by dengziming — APPROVED** [review]
LGTM.

**Comment by ex172000** [comment]
@satishd can you help to review?

**Review by satishd — COMMENTED** [review]
@ex172000 Thanks for the PR. I see that this is already committed. Reviewed the changes, LGTM.

**Pull request merged**